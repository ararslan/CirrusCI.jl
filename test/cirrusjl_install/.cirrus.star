load("../../lib.star", "cirrusjl_install")
load("github.com/cirrus-modules/helpers", "task", "windows_container", "freebsd_instance",
     "macos_instance", "container", "arm_container")

def _task(name, instance):
    os = name.split(" ")[0]
    return task(name, instance, instructions=[cirrusjl_install(os)])

def main(ctx):
    return [_task("FreeBSD", freebsd_instance("freebsd-13-0-release-amd64")),
            _task("Linux AArch64", arm_container("ubuntu:latest")),
            _task("Linux musl", container("alpine:3.14")),
            _task("Windows", windows_container("cirrusci/windowsservercore")),
            _task("macOS", macos_instance("big-sur-xcode"))]
