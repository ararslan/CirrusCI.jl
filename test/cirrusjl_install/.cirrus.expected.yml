task:
  name: FreeBSD
  freebsd_instance:
    image_family: freebsd-13-0
  install_script:
    - sh -c "$(fetch https://raw.githubusercontent.com/ararslan/CirrusCI.jl/master/bin/install.sh -o -)"
task:
  name: Linux AArch64
  arm_container:
    image: ubuntu:latest
    cpu: 2
    memory: 4096
  install_script:
    - sh -c "$(wget https://raw.githubusercontent.com/ararslan/CirrusCI.jl/master/bin/install.sh -q -O-)"
task:
  name: Linux musl
  container:
    image: alpine:3.14
    cpu: 2
    memory: 4096
  install_script:
    - sh -c "$(wget https://raw.githubusercontent.com/ararslan/CirrusCI.jl/master/bin/install.sh -q -O-)"
task:
  name: Windows
  windows_container:
    image: cirrusci/windowsservercore
  install_script:
    - ps: iex ((New-Object System.Net.WebClient).DownloadString('https://raw.githubusercontent.com/ararslan/CirrusCI.jl/master/bin/install.ps1'))
task:
  name: macOS
  macos_instance:
    image: big-sur-xcode
  install_script:
    - sh -c "$(curl https://raw.githubusercontent.com/ararslan/CirrusCI.jl/master/bin/install.sh)"
